create_rsna_infrastructure.py

```diff
@@ -40,31 +45,53 @@
 
 if args.ingress:
-    add_ingress_to_sg(ec2_client, STACK_NAME, vpc, "0.0.0.0/0", 8080, 8080)
+
+    print("[create_rsna_infrastructure] - Clear all existing Ingress.")
+    clear_sg_ingresses(ec2_client, STACK_NAME)
+
+    print("[create_rsna_infrastructure] - Adding Ingress.")
+    add_ingress_to_sg(STACK_NAME, vpc, "0.0.0.0/0", 8080, 8080)
+    add_ingress_to_sg(STACK_NAME, vpc, "0.0.0.0/0", 80, 80)
+    add_ingress_to_sg(STACK_NAME, vpc, "0.0.0.0/0", 443, 443)
 
 if args.ec2:
     create_ec2(ec2, settings, security_group, STACK_NAME)
 
 if args.elasticip:
     create_assign_elastic_ip(STACK_NAME, ec2_client)
 
-if args.deploy:
+if args.set_up_ec2:
+    print("[create_rsna_infrastructure] - Setting up EC2.")
     send_commands(STACK_NAME, ec2_client, ssm_client, ["sudo yum update -y && sudo yum install -y git "
-                                                   "&& sudo yum install -y docker "
-                                                   "&& sudo service docker restart "
-                                                   "&& sudo usermod -a -G docker ec2-user"
-                                                   "&& sudo curl -L https://github.com/docker/compose/releases/download/1.18.0/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose"
-                                                   "&& sudo chmod +x /usr/local/bin/docker-compose"
-                                                   "&& rm -Rf /home/ec2-user/s4s-stack"
-                                                   "&& git clone --recursive https://github.com/RSNA/s4s-stack /home/ec2-user/s4s-stack"
-                                                   "&& cd /home/ec2-user/s4s-stack"
-                                                   "&& git submodule update --init --recursive"
-                                                   "&& docker-compose down"
-                                                   "&& docker volume prune -f"
-                                                   "&& docker-compose up -d --force-recreate"])
+                                                       "&& sudo yum install -y docker "
+                                                       "&& sudo service docker restart "
+                                                       "&& sudo usermod -a -G docker ec2-user"
+                                                       "&& sudo curl -L https://github.com/docker/compose/releases/download/1.18.0/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose"
+                                                       "&& sudo chmod +x /usr/local/bin/docker-compose"
+                                                       "&& rm -Rf /home/ec2-user/reference-stack-docker"
+                                                       "&& git clone --recursive -b 152 https://github.com/sync-for-science/reference-stack-docker /home/ec2-user/reference-stack-docker"
```

Comment:
Can we factor out this magic branch number  (152)? I was super confused about how this worked.

Reply:
I.e. should merge into master without changing the default images that run in the master stack  -- and should name the directory to include the word "imaging" (rather than "s4s-stack").

Reply:
I have to jog my memory about what I was doing in that branch, I'll incorporate your suggestions though.